hip-362-token-freeze-amount.md

```diff
@@ -0,0 +1,44 @@
+**hip: TBD </br>
+title: Integrate '.setTokenAmount()' custom function for 'TokenFreezeTransaction()' </br>
+author: Tomachi Anura [shibartoken@protonmail.com](mailto:shibartoken@protonmail.com)</br>
+type: Standards Track</br>
+category: Service</br>
+needs-council-approval: Yes</br>
+status: Draft</br>
+last-call-date-time:</br>
+created: 22.02.13</br>
+discussions-to:**</br>
+
+## Abstract
+
+When "locking" or "freezing" liquidity related to an associated Account Id, There is no way to freeze a specific amount of tokens. The Shibar Network Team strongly believes that it would bring many advantages to have the tokenFreezeTransaction like in the sequent example:
+
+<img width="711" alt="Screen Shot 2022-02-13 at 5 44 48 PM" src="https://user-images.githubusercontent.com/96840872/153778591-6ba9b8b7-92a7-4438-b9fb-23e77a469e4d.png">
+
+## Motivation
+
+When "locking" or "freezing" liquidity related to an associated Account Id, There is no way to freeze a specific amount of tokens. This could be potentially detrimental in regards of DeFi, if the same token offers many use-cases. As it stands now, Users would have to set different accounts in order to use the same Network.
+
+
+## Rationale
+When Alice participates to a round 1 token pre-sale, she buys 1000 tokens. According to the terms and conditions of the round 1 pre-sale, she should have those tokens frozen and unable to move them from the wallet until the term expires. However Alice would also like to buy more tokens at round 2 which are supposed to be Unfrozen tokens. Unfortunately Hedera only allows to freeze the whole token amount of the account, while it would be much more beneficial to implement a child function .setTokenAmount(Amount) to the parent TokenFreezeTransaction().
+
+This implementation allows Alice to interact with the token if she holds more than the frozen amount. for many use cases, without limiting her whole account by keeping the whole token amount frozen. This way, she can buy another 1000 tokens on round 2, having a total balance of 2000. And she is able to use unfrozen 1000 tokens from round 2 to interact with the network
```

Comment:
```suggestion
This implementation allows Alice to interact with the token if she holds more than the frozen amount. for many use cases, without limiting her whole account by keeping the whole token amount frozen. This way, she can buy another 1000 tokens on round 2, having a total balance of 2000 and she can use unfrozen 1000 tokens from round 2 to interact with the network
```

Reply:
DCO check is failing!

Reply:
https://github.com/hashgraph/hedera-improvement-proposal/pull/395/commits/5b43f814c042f8bab8eeacd474ab6380f30fdaea